### PR TITLE
Restore metadata support to generated clients

### PR DESCRIPTION
**Pull request by lowasser**
Addresses #224.  Users generally shouldn't use this, but it's hard to get any other way, so let's put it back in.

### CODE REVIEW

**Comment by dashingsat** [comment]
Any idea on when this will be merged..?

**Comment by jamesward** [comment]
@lowasser Want @bshaffer or I to review this?

**Comment by lowasser** [comment]
Yes, please.

File: compiler/src/main/java/io/grpc/kotlin/generator/GrpcClientStubGenerator.kt [hunk 1]
```diff
@@ -193,7 +199,7 @@ class GrpcClientStubGenerator(config: GeneratorConfig) : ServiceCodeGenerator(co
         %methodDescriptor:L,
         %parameter:N,
         callOptions,
-        %headers:T()
+        %headers:N
```
Comment (bshaffer) [inline compiler/src/main/java/io/grpc/kotlin/generator/GrpcClientStubGenerator.kt:202]: potentially dumb question, buy why is the headers and parameters placeholder the same (`:N`)?

**Review by bshaffer — APPROVED** [review]
LGTM

**Pull request merged**